tests/integration/acp.go

```diff
@@ -120,6 +120,119 @@ func addPolicyACP(
 	}
 }
 
+// AddDocActorRelationship will attempt to add the given policy using DefraDB's ACP system.
+type AddDocActorRelationship struct {
+	// NodeID may hold the ID (index) of the node we want to add doc actor relationship on.
+	//
+	// If a value is not provided the relationship will be added in all nodes, unless testing with
+	// sourcehub ACP, in which case the relationship will only be defined once.
+	NodeID immutable.Option[int]
+
+	// The collection in which this document we want to add a relationship for exists.
+	//
+	// This is a required field, to test the invalid usage of not having this arg, use -1 index.
+	CollectionID int
+
+	// The index-identifier of the document within the collection.  This is based on
+	// the order in which it was created, not the ordering of the document within the
+	// database.
+	//
+	// This is a required field, to test the invalid usage of not having this arg, use -1 index.
+	DocID int
+
+	// The name of the relation to set between document and target actor (within the policy).
+	Relation string
+
+	// The target public identity, i.e. the identity of the actor to tie the document's relation with.
+	//
+	// This is a required field, to test the invalid usage of not having this arg, use -1 index.
+	TargetIdentity int
+
+	// The requestor identity, i.e. identity of the actor creating the relationship.
+	// Note: This identity must either own or have managing access defined in the policy.
+	//
+	// This is a required field, to test the invalid usage of not having this arg, use -1 index.
+	RequestorIdentity int
+
+	// Result returns true if it was a no-op due to existing before, and false if a new relationship was made.
+	ExpectedExistence bool
+
+	// Any error expected from the action. Optional.
+	//
+	// String can be a partial, and the test will pass if an error is returned that
+	// contains this string.
+	ExpectedError string
+}
+
+func addDocActorRelationshipACP(
+	s *state,
+	action AddDocActorRelationship,
+) {
+	for i, node := range getNodes(action.NodeID, s.nodes) {
+		var collectionName string
+		if action.CollectionID == -1 {
+			collectionName = ""
+		} else {
+			collection := s.collections[i][action.CollectionID]
+			if !collection.Description().Name.HasValue() {
+				require.Fail(s.t, "Expected non-empty collection name, but it was empty.", s.testCase.Description)
+			}
+			collectionName = collection.Description().Name.Value()
+		}
+
+		var docID string
+		if action.DocID == -1 {
```

Comment:
discussion: I understand if people prefer having `DocIDIndex` and `DocID` to test these edge cases rather than `-1` to test empty/invalid DocID input. I don't have any strong preferences here, happy to follow consensus.

Reply:
`immutable.Option` is for optional fields. This is not an optional field, it's more to test invalid input of that field. Is why I was wondering if others prefer non-index input as well (i.e. `DocIDIndex` for normal cases and `DocID` for actual invalid testing).